### ISSUE TITLE
Run one sample per node

### ISSUE THREAD

**Issue by oskarvid** [elixir-no-nels/Selma#5]
The current version will run all samples on one execute node, this leads to potential issues that are resolved by running one sample per node.

**Comment by oskarvid** [comment]
A one-sample-per-node version is currently being tested, contingency actions for failure cleanup and clean up after finishing a job has been added and works so far.  
Will continue to test before closing this issue.

**Comment by oskarvid** [comment]
Function has been thoroughly tested and works as intended. Closing issue.

**Comment by kjellp** [comment]
If multiple samples are run, and the staging of input files for a second sample fails (typo in paths/names in tsv file forinstance), it seems like:
- already submitted jobs will still stay in queue and execute when resources become available
- the complete staging directory for all samples in this execution is removed (also for the Samples without errors?), or is it only for the failing job the staging dir is removed?
- the script hangs after saying "[Info] Exited on <timestamp>"\n"[ERROR]Execution stopped unexpectedly, here's the logfile:..."

Can be a case of confusing multiple threads of output. Will see if it completes successfully for the sample with ok input files, and that output files are successfully transferred back for that sample.

**Comment by oskarvid** [comment]
1. Yes
2. The failing job's staging directory is removed, this happens when [this](https://github.com/elixir-no-nels/Selma/blob/milestone2/scripts/functions.sh#L119) variable in the function is assigned the unique name of the staging directory for each sample. The [rmtmp](https://github.com/elixir-no-nels/Selma/blob/milestone2/scripts/functions.sh#L237) function then removes the staging directory for that job if it fails.
3. I ran a test just now and it may look like the script hangs but it keeps going once the remaining jobs start running.

I found a bug myself though, if you name the samples something like "sample, sample2, sample3, etc" it wouldn't select only sample or sample2, sample3, etc because the grep function that does this selection didn't have the -w flag to enable exact matches only. It works as it should now.
